hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -869,4 +869,26 @@ public void notifyLeaderChanged(RaftGroupMemberId groupMemberId,
                                   RaftPeerId raftPeerId) {
     ratisServer.handleLeaderChangedNotification(groupMemberId, raftPeerId);
   }
+
+  @Override
+  public String toStateMachineLogEntryString(StateMachineLogEntryProto proto) {
+    try {
+      ContainerCommandRequestProto requestProto =
+              getContainerCommandRequestProto(proto.getLogData());
+      long contId = requestProto.getContainerID();
+
+      switch (requestProto.getCmdType()) {
+      case WriteChunk:
+        String location = containerController.getContainerLocation(contId);
+        return HddsUtils.writeChunkToString(requestProto.getWriteChunk(),
+                contId, location);
+      default:
+        throw new IllegalStateException("Cmd Type:" + requestProto.getCmdType()
+                + " should not have state machine data");
+
+      }
+    } catch (Throwable t) {
+      return "";
```

Comment:
We will catch the IllegalStateException here.

Reply:
The function to LogEntryString does not expect an exception to be thrown, should I return an empty string at 886 instead ?